<!--
  Copyright 2016 The Cartographer Authors
  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at
       http://www.apache.org/licenses/LICENSE-2.0
  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
-->
<robot name="mini">
  <material name="orange">
    <color rgba="1.0 0.5 0.2 1" />
  </material>
  <material name="gray">
    <color rgba="0.2 0.2 0.2 1" />
  </material>
 
  <link name="base_link">
    <visual>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.60 0.60 0.2" />
      </geometry>
      <material name="orange" />
    </visual>
  </link>    

  <link name="imu_frame"> 
    <visual>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.05 0.05 0.05" />
      </geometry>
      <material name="orange" />
    </visual>
  </link>    

  <link name="laser_frame"> 
    <visual>
      <origin xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.03" />
      </geometry>
      <material name="gray" />
    </visual>
  </link>
 
  <joint name="laser2imu" type="fixed">
    <parent link="laser_frame" />
    <child link="imu_frame" />
    <origin xyz="-0.23 0.14 -0.05 " rpy="0 0 -1.5707"/>
  </joint>

  <joint name="imu2base_link" type="fixed">
    <parent link="imu_frame" />
    <child link="base_link" />
    <origin xyz="0.23 0 0 " rpy="0 0 3.1415926" />
  </joint>  
</robot>
